### ISSUE TITLE
[FL autosuggest prototype] The location input errors on blur instead of on submit

### ISSUE THREAD

**Issue by qjac** [department-of-veterans-affairs/va.gov-cms#22458]
## Describe the defect
Found as part of [ADE accessibility audit](https://github.com/department-of-veterans-affairs/va.gov-team/issues/120537#issuecomment-3347814120) of the location search box, Facility type select, custom select, and autosuggest fields on the [autosuggest prototype](https://staging.va.gov/find-locations/).

> **Error triggers on blur**
> **Field:** Location search box
> **Expected severity level:** 4
> **Expected experience standard:** User experiences predictable interactions with components and patterns
> **Description:** The “Zip code or city, state” field’s error triggers when the user tabs out of the field, not when they press “Search."
> - This can frustrate users, especially those using AT, where data entry might take extra time.
> - It’s also not consistent with the current version of the VADS text input component.
> **Recommendation:** The text input isn’t a VADS component. If it’s possible to convert it to a [va-text-input](https://design.va.gov/storybook/?path=/docs/uswds-va-text-input--docs), do that - you’ll get the proper error handling for free. If that’s not possible, review how va-text-input handles errors - if the user hasn’t typed into the field, the error shouldn’t trigger.

[Location input is a custom combox](https://dsva.slack.com/archives/C0FQSS30V/p1759523788832689?thread_ts=1759522358.548369&cid=C0FQSS30V) and is therefore not inheriting the error handling from VADS inputs. We want to match how [va-text-input](https://design.va.gov/components/form/text-input) handles errors - if the user hasn’t typed into the field, the error shouldn’t trigger on blur. The error should only trigger on submitting 'Search' and focus shifted to the first `va-text-input` in the form with an error.

[VADS mock form showing desired implementation](https://staging.va.gov/mock-simple-forms-patterns/text-input)
[VADS form error handling guidance](https://design.va.gov/components/form/#error-handling)
[Additional VADS required input handling guidance](https://design.va.gov/components/form/#what-to-do-instead)

## To Reproduce
1. Go to [autosuggest prototype](https://staging.va.gov/find-locations/)
2. TAB into, then out of, the location field, without typing anything
3. See the error "Enter a zip code or a city and state in the search box" immediately after leaving the field

## AC / Expected behavior
- [ ] The user can tab through form without errors displaying even if required fields remain empty.
- [ ] If required fields remain empty when the user clicks 'Search, then display the error message and move focus to first input with an error.
- [x] Requires accessibility review

## Definition of done
- [x] Code completed; PR created
- [x] Review instance created/available; internal team notified to start reviews
- [ ] Code review (meets established quality and security standards)
- [ ] Design review (meets design & UX requirements)
- [x] A11y review (meets accessibility standards)
- [x] Functional/feature review (meets acceptance criteria)
- [ ] Automated tests passed; PR approved
- [x] Testing completed; dev team notified to deploy
- [ ] PO/PM/external verify in prod
- [ ] Documentation updated

**Comment by carlyMhayter** [comment]
>  But other validations could error at that point, like a formatting error if the field needs to be formatted like an email or if a minimum number of characters is expected. Those types of 'format validation' should error on blur if a value has been set.

I see that makes sense! So for the location input, is there an expectation that that text is formatted a particular way? Right now, if a user attempts to click/tab out of the input and the text entered is not a valid zip code, or city/state, there isn't an error.

For I will address the **empty** field erroring on blur but I just wanted to flag the formatting validation as well!

**Comment by r-bartlett-gsa** [comment]
@carlyMhayter Thank you for flagging that!! We do need to have formatting validation for the location field, and I will create a ticke for that. 
cc: @qjac

**Comment by qjac** [comment]
Reviewed https://github.com/department-of-veterans-affairs/vets-website/pull/40139 in Edge (on CAG). And a 'required' error in the location no longer fires on blur, but does on submit. :+1::skin-tone-2:

**Comment by r-bartlett-gsa** [comment]
This is blocked because of the inability to preview the code on an RI.
